app/attributes/attributecontroller.h

```diff
@@ -80,6 +80,10 @@ class  AttributeController : public QObject
     //! Returns TRUE if all soft constraints defined on fields are satisfied with the current attribute values
     Q_PROPERTY( bool constraintsSoftValid READ constraintsSoftValid NOTIFY constraintsSoftValidChanged )
 
+    //! Returns TRUE if all hard constraints are satisfied and all fields have valid state.
+    //! Validity of a field depends on its type, f.e. numeric field is valid when input is convertible to number and number is in correct min/max range
```

Comment:
I am not sure if it is better to combine or not combine hard constrains and invalid state into single boolean. we may have bool property to evaluate if all values are valid. and in qml/c++ you can do contrainsHardValid && allFieldsValidValues.

just for consideration

Reply:
Changed it to have separate boolean for values validity, #1507 will join them to `errorMessage`